src/components/layout/GenericLayout/index.tsx

```diff
@@ -1,19 +1,41 @@
 import React from 'react'
-import { Header } from './Header'
+
+import ThemeProvider, { StaticGlobalStyle, ThemedGlobalStyle } from 'theme'
+
 import { Footer } from './Footer'
-import { StaticGlobalStyle } from 'theme'
-export interface Props {
-  menu?: React.ReactNode
-  navTools?: React.ReactNode
+import { Header } from './Header'
+
+export type Props = {
+  header?: JSX.Element | null
+  footer?: JSX.Element | null
+  children: React.ReactNode
 }
 
-export const GenericLayout: React.FC<Props> = ({ children, menu, navTools }) => (
-  <div>
-    <StaticGlobalStyle />
-    <Header menu={menu} tools={navTools} />
-    {children}
-    <Footer />
-  </div>
-)
+const defaultHeader = <Header />
+const defaultFooter = <Footer />
+
+/**
+ * Generic layout with optional header and footer
+ * Applies global and theme styles to all children
+ *
+ * If not header/footer set, use default.
+ * If a custom passed in will be used instead.
+ * To remove header/footer, pass null
+ */
+export const GenericLayout: React.FC<Props> = (props) => {
```

Comment:
```suggestion
export const GenericLayout: React.FC<Props> = ({
  header = defaultHeader,
  footer = defaultFooter,
  children,
}) => {
```

Reply:
The problem with this is that it'll overwrite `null`, won't it?

Reply:
I was wrong, it does work. I'll take it.